tests/server_root/router.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+$file = __DIR__ . $_SERVER['SCRIPT_NAME'];
```

Comment:
This file was added to support the functional testing. PHP's built-in server doesn't natively handle `If-Modified-Since` headers, so I had to do implement that here.